### PR TITLE
Add filtering for calendars.txt start_date and end_date to web app

### PR DESCRIPTION
**Pull request by owades**
- Filter based on logic in GTFS spec
- Filters can be overrided by ignore-start-end-date in agency-params.json
- Test value can be passed in via utm param: testdate
- This PR also removes two unused features: driver-select dropdown and "mode"

For a test agency, I confirmed:
- Filtering will show a trip if the current date is the start date or end date
- Filtering will not show a trip if current date is before start, or after end
- Filtering will show all trips if ignore-start-end-date is set to true (though, cache doesn't refresh immediately)

### CODE REVIEW

File: server/app-engine/static/graas.js [hunk 1]
```diff
@@ -90,6 +87,19 @@ function getDayOfWeek() {
...
+        return testDate;
+    } else {
+        var today = new Date();
+        var year = today.getFullYear();
+        var month = today.getMonth() + 1;
+        var date = today.getDate();
+        return (year * 10000) + (month * 100) + date;
+    }
+}
+
```
Comment (unreasonableman) [inline server/app-engine/static/graas.js:102]: gtfs-rt-util.js has some time and date utility functions. Consider adding this there.

**Review by unreasonableman — APPROVED** [review]

File: server/app-engine/static/graas.js (hunk 1 above)
Comment (owades) [reply server/app-engine/static/graas.js]: updated - mind taking one more quick look? Tested and is working fine

File: server/app-engine/static/gtfs-rt-util.js [hunk 2]
```diff
@@ -89,6 +89,63 @@ if (!fetch) {
...
+
+    // returns 0 for Monday...and 6 for Sunday
+    exports.getDayOfWeek = function() {
+        if (testDow) {
+            return testDow;
+        } else return ((new Date()).getDay() + 6) % 7;
+    }
+
+    // returns date as an 8-character string (ie 20220317 for 3/17/22)
+    exports.getDate = function() {
```
Comment (unreasonableman) [inline server/app-engine/static/gtfs-rt-util.js:101]: gtfs-rt-util.js already has a getDate() function (that does something different). Choose a different name, or else one function will be permanently hidden.

**Review by unreasonableman — CHANGES_REQUESTED** [review]
please see comment

**Review by unreasonableman — APPROVED** [review]

**Pull request merged**